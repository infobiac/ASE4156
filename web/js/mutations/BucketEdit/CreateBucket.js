// @flow
import { graphql, commitMutation } from 'react-relay';

import type {
  RecordSourceSelectorProxy,
  RelayEnvironment,
  SelectorData,
} from 'react-relay';

import type {
  CreateBucketMutationVariables,
  CreateBucketMutationResponse,
} from './__generated__/CreateBucketMutation.graphql';

const mutation = graphql`
  mutation CreateBucketMutation($name: String!, $public: Boolean!, $investment: Float!) {
    addBucket(name: $name, public: $public, investment: $investment) {
       bucket {
         id
         ...InvestBucketRelay_bucket
      }
    }
  }
`;

export default (
  updater?: ?(store: RecordSourceSelectorProxy, data: SelectorData) => void,
  optimisticUpdater?: ?(store: RecordSourceSelectorProxy) => void,
  onCompleted?: ?(response: ?CreateBucketMutationResponse, errors: ?[Error]) => void,
) => (environment: RelayEnvironment) => (variables: CreateBucketMutationVariables) => {
  const optimisticResponse = {
    addBucket: {
      bucket: {
        name: variables.name,
        available: variables.investment,
        description: {
          edges: [],
        },
      },
    },
  };
  commitMutation(environment, {
    mutation,
    variables,
    updater,
    optimisticResponse,
    optimisticUpdater,
    onCompleted,
  });
};
