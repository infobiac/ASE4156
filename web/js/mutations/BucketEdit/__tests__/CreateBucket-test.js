import {
  RecordSource,
  Store,
  Environment,
} from 'relay-runtime';
import { commitMutation } from 'react-relay';
import CreateBucket from '../CreateBucket';

jest.mock('react-relay', () => ({
  graphql: jest.fn(),
  commitMutation: jest.fn(),
}));

const ctx = () => {
  const source = new RecordSource();
  const store = new Store(source);
  const network = jest.fn();
  return {
    environment: new Environment({
      network,
      store,
    }),
    variables: {},
  };
};

describe('CreateBucket', () => {
  it('exists', () => {
    expect(CreateBucket).toBeTruthy();
  });

  it('mutates', () => {
    const data = { investment: 3, name: 'name', public: false };
    CreateBucket(null, null, null)(ctx().environment)(data);
    expect(commitMutation.mock.calls.length).toEqual(1);
    expect(commitMutation.mock.calls[0][1].variables).toEqual(data);
    expect(commitMutation.mock.calls).toMatchSnapshot();
  });
});
