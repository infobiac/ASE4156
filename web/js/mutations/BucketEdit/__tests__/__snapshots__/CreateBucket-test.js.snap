// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateBucket mutates 1`] = `
Array [
  Array [
    RelayModernEnvironment {
      "__setNet": [Function],
      "_debugger": EnvironmentDebugger {
        "_envIsDirty": false,
        "_environment": [Circular],
        "_id": "RelayModernEnvironment1",
        "_isRecordingMutationEvents": false,
        "_recordedMutationEvents": Array [],
      },
      "_network": [Function],
      "_publishQueue": RelayPublishQueue {
        "_appliedOptimisticUpdates": Set {},
        "_backup": Object {},
        "_handlerProvider": [Function],
        "_pendingBackupRebase": false,
        "_pendingData": Set {},
        "_pendingOptimisticUpdates": Set {},
        "_pendingUpdaters": Set {},
        "_store": RelayMarkSweepStore {
          "_hasScheduledGC": false,
          "_index": 0,
          "_recordSource": Object {},
          "_roots": Map {},
          "_subscriptions": Set {},
          "_updatedRecordIDs": Object {},
        },
      },
      "_store": RelayMarkSweepStore {
        "_hasScheduledGC": false,
        "_index": 0,
        "_recordSource": Object {},
        "_roots": Map {},
        "_subscriptions": Set {},
        "_updatedRecordIDs": Object {},
      },
      "configName": undefined,
      "unstable_internal": Object {
        "areEqualSelectors": [Function],
        "createFragmentSpecResolver": [Function],
        "createOperationSelector": [Function],
        "getDataIDsFromObject": [Function],
        "getFragment": [Function],
        "getOperation": [Function],
        "getSelector": [Function],
        "getSelectorList": [Function],
        "getSelectorsFromObject": [Function],
        "getVariablesFromObject": [Function],
      },
    },
    Object {
      "mutation": [Function],
      "onCompleted": null,
      "optimisticResponse": Object {
        "addBucket": Object {
          "bucket": Object {
            "available": 3,
            "description": Object {
              "edges": Array [],
            },
            "name": "name",
          },
        },
      },
      "optimisticUpdater": null,
      "updater": null,
      "variables": Object {
        "investment": 3,
        "name": "name",
        "public": false,
      },
    },
  ],
]
`;
